gammapy/scripts/spectrum_pipe.py

```diff
@@ -0,0 +1,220 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (print_function)
+import logging
+from ..utils.scripts import get_parser, set_up_logging_from_args
+from ..obs import DataStore, ObservationTable
+from ..irf import EnergyDispersion, EnergyDispersion2D
+from ..irf import EffectiveAreaTable, EffectiveAreaTable2D
+from ..data import CountsSpectrum, EventList
+from ..spectrum import EnergyBounds, Energy
+from astropy.coordinates import Angle, SkyCoord
+import sherpa.astro.ui as sau
+from gammapy.hspec import wstat
+from sherpa.models import PowLaw1D
+from gammapy.background import ring_area_factor
+import numpy as np
+
+__all__ = ['GammapySpectrumAnalysis', 'GammapySpectrumObservation']
+
+log = logging.getLogger(__name__)
+
+
+def main(args=None):
+    parser = get_parser(GammapySpectrumAnalysis)
+    parser.add_argument('config_file', type=str,
+                        help='Config file in YAML format')
+    parser.add_argument("-l", "--loglevel", default='info',
+                        choices=['debug', 'info', 'warning', 'error', 'critical'],
+                        help="Set the logging level")
+    args = parser.parse_args(args)
+    set_up_logging_from_args(args)
+    analysis = GammapySpectrumAnalysis.from_yaml(args.config_file)
+    analysis.make_ogip()
+    analysis.run_fit()
+
+class GammapySpectrumAnalysis(object):
+    """Entrance points for gammapy-spectrum-pipe
+    """
+    def __init__(self, config):
+        self.config = config
+        vals = config['general']['observations']
+        if isinstance (vals, basestring):
+            vals = np.loadtxt(vals, dtype=np.int)
+        self.obs = vals[0]
+        _process_config(self)
+        self.info()
+        self.observations = []
+        for obs in vals:
+            val = GammapySpectrumObservation(obs, config)
+            self.observations.append(val)
+
+    @classmethod
+    def from_yaml(cls, filename):
+        """Read config from YAML file."""
+        import yaml
+        log.info('Reading {}'.format(filename))
+        with open(filename) as fh:
+            config = yaml.safe_load(fh)
+        return cls(config)
+
+    def info(self):
+        log.info(self.store.info())
+        log.info('ON region\nCenter: {0}\nRadius: {1}'.format(
+            self.target,self.radius))
+        log.info('OFF region\nType: {0}\nInner Radius: {1}\nOuter Radius: {2}'.format(
+            'Ring',self.irad,self.orad))
+
+    def make_ogip(self):
+        for obs in self.observations:
+            log.info('Creating OGIP data for run{}'.format(obs.obs))
+            obs.make_ogip()
+
+    def run_fit(self):
+        log.info("Starting HSPEC")
+
+        if (self.model == 'PWL'):
+            p1 = PowLaw1D('p1')
+            p1.gamma = 2.2
+            p1.ref = 1e9
+            p1.ampl = 6e-19
+        else:
+            log.error('Desired Model is not defined')
```

Comment:
It probably doesn't make sense to continue here, because you'll just raise a more cryptic error below when p1 isn't defined. So you should raise a ValueError here instead of emitting a log.error message.